### PR TITLE
Scala future baseline tests

### PR DESCRIPTION
**Pull request by GDownes**
### Overview
Series of tests to establish baseline agent behavior with and without instrumentation for Scala Futures

### Checks

[x] Are your contributions backwards compatible with relevant frameworks and APIs? Yes
[x] Does your code contain any breaking changes? No
[x] Does your code introduce any new dependencies? No

### CODE REVIEW

File: instrumentation/scala-2.13.0/src/test/scala/com/nr/agent/instrumentation/scala/baseline/InstrumentedSegmentNestedAsynchronousMultipleThreadTest.scala [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+  @Trace(dispatcher = true)
+  def getTwoResults: Future[Int] = Future.reduceLeft(Seq(getFirstNumber, getSecondNumber))(_ + _)(threadPoolTwo)
+
+  @Trace(dispatcher = true)
+  def getThreeResults: Future[Int] = Future.reduceLeft(Seq(getFirstNumber, getSecondNumber, getThirdNumber))(_ + _)(threadPoolThree)
+
+  @Trace
+  def getFirstNumber: Future[Int] = Future{
+    println(s"${Thread.currentThread.getName}: getFirstNumber")
```
Comment (meiao) [inline instrumentation/scala-2.13.0/src/test/scala/com/nr/agent/instrumentation/scala/baseline/InstrumentedSegmentNestedAsynchronousMultipleThreadTest.scala:92]: These println calls look like were used for debugging the tests.
But I think it is going to be just noise when the tests run normally.

**Review by meiao — APPROVED** [review]
If the println does provide value, this can be merged.

File: instrumentation/scala-2.13.0/src/test/scala/com/nr/agent/instrumentation/scala/baseline/InstrumentedSegmentNestedAsynchronousMultipleThreadTest.scala (hunk 1 above)
Comment (GDownes) [reply instrumentation/scala-2.13.0/src/test/scala/com/nr/agent/instrumentation/scala/baseline/InstrumentedSegmentNestedAsynchronousMultipleThreadTest.scala]: Removed println calls

**Review by meiao — APPROVED** [review]
LGTM

**Pull request merged**